consensus/parlia/parlia.go

```diff
@@ -1850,13 +1889,32 @@ func (p *Parlia) getCurrentValidators(blockHash common.Hash, blockNum *big.Int)
 	return valSet, voteAddrMap, nil
 }
 
+func (p *Parlia) isBlockDelayMinedOnPurpose(chain core.ChainContext, header *types.Header) (bool, error) {
```

Comment:
this check seems useless, it is easy to workaround, validator can always assign `header.MilliTimestamp() <= parent.MilliTimestamp()+blockInterval`

Reply:
when validator can always assign header.MilliTimestamp() <= parent.MilliTimestamp()+blockInterval,
then they delay to block broadcast, if the delay is enough, back up validator will produce the blocks and slash the in-turn validator.

the check here is to ensure `header.MilliTimestamp() = parent.MilliTimestamp()+blockInterval`